extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcIdentityProvider.java

```diff
@@ -196,6 +184,22 @@ public Uni<SecurityIdentity> apply(TokenVerificationResult result, Throwable t)
                 });
     }
 
+    private Uni<TokenVerificationResult> verifyPrimaryTokenUni(Map<String, Object> requestData,
+            TokenAuthenticationRequest request, TenantConfigContext resolvedContext, UserInfo userInfo) {
+        if (isInternalIdToken(request)) {
+            if (requestData.get(NEW_AUTHENTICATION) == Boolean.TRUE) {
```

Comment:
This may not evaluate to true if the left call was created using `new Boolean(...)`

Reply:
Thanks @gastaldi, yes, it won't happen, `CodeFlowAuthenticationMechanism` sets it [here](https://github.com/quarkusio/quarkus/blob/main/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java#L859)

Reply:
@gastaldi I may refactor it a bit later anyway :-), right now this particular code can't really work independently from CodeFlowAuthenticationMechanism, so it is safe for now